src/vault/sys/wrapping.clj

```diff
@@ -0,0 +1,90 @@
+(ns vault.sys.wrapping
+  "The /sys/wrapping endpoint is used to wrap secrets and lookup, rewrap, and
+  unwrap tokens.
+
+  Reference:
+  - https://www.vaultproject.io/api-docs/system/wrapping-lookup
+  - https://www.vaultproject.io/api-docs/system/wrapping-rewrap
+  - https://www.vaultproject.io/api-docs/system/wrapping-unwrap
+  - https://www.vaultproject.io/api-docs/system/wrapping-wrap"
+  (:require
+    [vault.client.http :as http]
+    [vault.util :as u])
+  (:import
+    vault.client.http.HTTPClient))
+
+
+;; ## API Protocol
+
+(defprotocol API
+  "The wrapping endpoint is used to manage response-wrapped tokens."
+
+  (lookup
+    [client token-id]
+    "Returns the wrapping properties for the given token.")
+
+  (rewrap
+    [client token-id]
+    "Rotates the given wrapping token and refreshes its TTL. Returns the new
+    token info.")
+
+  (unwrap
+    [client token-id]
+    "Returns the original response inside the given wrapping token.")
+
+  (wrap
+    [client data ttl]
+    "Wraps the given map of data inside a response-wrapped token with the
+    specified TTL. The TTL can be either an integer number of seconds or a
+    string duration of seconds (15s), minutes (20m), or hours (25h). Returns the
+    new token info."))
+
+
+(defn- kebabify-body-wrap-info
+  [body]
+  (u/kebabify-keys (get body "wrap_info")))
+
+
+;; ## HTTP Client
```

Comment:
Really minor nit: this helper function is for the HTTP code rather than part of the protocol, so:
```suggestion
;; ## HTTP Client

(defn- kebabify-body-wrap-info
  [body]
  (u/kebabify-keys (get body "wrap_info")))

```